prov/shm/src/smr_util.c

```diff
@@ -466,6 +475,23 @@ void smr_map_to_endpoint(struct smr_region *region, int64_t id)
 	if ((region != peer_smr && region->cma_cap_peer == SMR_VMA_CAP_NA) ||
 	    (region == peer_smr && region->cma_cap_self == SMR_VMA_CAP_NA))
 		smr_cma_check(region, peer_smr);
+
+	/* enable xpmem locally if the peer also has it enabled */
+	if (peer_smr->xpmem_cap_self == SMR_VMA_CAP_ON &&
+	    region->xpmem_cap_self == SMR_VMA_CAP_ON) {
+		local_peers[id].xpmem.apid = xpmem_get(peer_smr->xpmem_self.seg_id,
+					XPMEM_RDWR, XPMEM_PERMIT_MODE, (void *) 0666);
+		if (local_peers[id].xpmem.apid == -1) {
+			local_peers[id].xpmem.cap = SMR_VMA_CAP_OFF;
+			return;
+		}
+		local_peers[id].xpmem.cap = SMR_VMA_CAP_ON;
+		local_peers[id].xpmem.addr_max = peer_smr->xpmem_self.address_max;
+	} else {
+		local_peers[id].xpmem.cap = SMR_VMA_CAP_OFF;
+	}
+
+	return;
```

Comment:
can you hide this in an xpmem specific function that compiles to a no op for the non xpmem case?